ui/src/app/pages/signed-in/component.ts

```diff
@@ -103,6 +112,42 @@ export class SignedInComponent implements OnInit, OnDestroy, AfterViewInit {
     cdrVersionsApi().getCdrVersions().then(resp => {
       cdrVersionStore.next(resp.items);
     });
+
+    this.startUserActivityTracker();
+    this.startInactivityTimers();
+  }
+
+  startUserActivityTracker() {
+    const signalUserActivity = debouncer(() => {
+      window.postMessage(INACTIVITY_CONFIG.MESSAGE_KEY, '*');
+    }, 1000);
+
+    INACTIVITY_CONFIG.TRACKED_EVENTS.forEach(eventName => {
+      window.addEventListener(eventName, () => signalUserActivity(), false);
+    });
+  }
+
+  startInactivityTimers() {
+    const resetLogoutTimeout = resettableTimeout(() => {
+      this.signOut();
+    }, environment.inactivityTimeoutInSeconds * 1000);
+
+    const resetInactivityModalTimeout = resettableTimeout(() => {
+      this.showInactivityModal = true;
+    }, (environment.inactivityTimeoutInSeconds - environment.inactivityWarningInSecondsBefore) * 1000);
+
+    localStorage.setItem(INACTIVITY_CONFIG.LOCAL_STORAGE_KEY_LAST_ACTIVE, Date.now().toString());
```

Comment:
One possible edge case here: there's nothing tying this local storage value to the current user login. Should we be somehow incorporating that into the key? This would only matter in a multilogin / shared machine situation

e.g. I could flip through multiple logins without having to reenter my password.

Reply:
Good point